frontend/src/pages/projects/screens/detail/notebooks/useNotebookDeploymentSize.ts

```diff
@@ -1,33 +1,38 @@
 import * as React from 'react';
-import * as _ from 'lodash';
 import { AppContext } from '../../../../../app/AppContext';
 import { NotebookContainer, NotebookSize } from '../../../../../types';
 import { getNotebookSizes } from '../../../../notebookController/screens/server/usePreferredNotebookSize';
 import { NotebookKind } from '../../../../../k8sTypes';
 
-const useNotebookDeploymentSize = (notebook?: NotebookKind): NotebookSize | null => {
+const useNotebookDeploymentSize = (
+  notebook?: NotebookKind,
+): { size: NotebookSize | null; error: string } => {
   const { dashboardConfig } = React.useContext(AppContext);
 
-  if (!notebook) {
-    return null;
-  }
-
   const container: NotebookContainer | undefined = notebook?.spec.template.spec.containers.find(
     (container) => container.name === notebook.metadata.name,
   );
 
   if (!container) {
-    return null;
+    return { size: null, error: 'Failed to get workbench information.' };
   }
 
   const sizes = getNotebookSizes(dashboardConfig);
-  const size = sizes.find((size) => _.isEqual(size.resources.limits, container.resources?.limits));
+  const size = sizes.find(
+    (size) =>
+      size.resources.limits?.cpu === container.resources?.limits?.cpu &&
+      size.resources.limits?.memory === container.resources?.limits?.memory,
+  );
 
   if (!size) {
-    return null;
+    return {
+      size: null,
+      error:
+        'Failed to get size information, the size might be changed after the workbench creation, please check your dashboard config.',
```

Comment:
@kywalker-rh thoughts on wording?

Reply:
I like the solution, thanks for putting that together @DaoDaoNoCode! I can't think of any changes, but tagging @kaedward in case she has feedback.

Reply:
If Katie would like any changes, we can do another PR. I think we have enough eyeballs on this to consider this a good first state.

Reply:
@andrewballantyne @kywalker-rh Does this happen when the workbench size is changed from the dashboard and it hasn't updated on the screen yet? If that's the case, I'd suggest something along the lines of "Workbench size is currently unavailable. Check your dashboard configuration to view size information, or refresh the page."

LMK if I'm not fully understanding the issue here!

Reply:
@kaedward It is when you create a Workbench -- then at some point after change the Notebook Size value in the ODHDashboardConfig -- since it's only the value we store in the Workbench, we reverse engineer the name of the Notebook Size by matching it 1:1 with the values for the resource limits. 

So it's not currently something they can configure in the Dashboard, but it is a logical state the code can get into.

Reply:
```suggestion
        'Workbench size is currently unavailable. Check your dashboard configuration to view size information.',
```
@DaoDaoNoCode Please do this change.